packages/ra-ui-materialui/src/button/DeleteWithUndoButton.tsx

```diff
@@ -23,8 +23,8 @@ const DeleteWithUndoButton: FC<DeleteWithUndoButtonProps> = props => {
         className,
         icon = defaultIcon,
         onClick,
-        resource,
-        record,
+        resource = '',
+        record = { id: '' },
```

Comment:
same, please do not use a default value

Reply:
I was using type-script with strictMode = true. I'll change it.